conjure-java-core/src/integrationInput/java/com/palantir/product/SafeExternalLongExample.java

```diff
@@ -0,0 +1,189 @@
+        if (result == 0) {
+            int hash = 1;
+            hash = 31 * hash + Long.hashCode(this.safeExternalLongValue);
+            hash = 31 * hash + this.optionalSafeExternalLong.hashCode();
+            result = hash;
+            memoizedHashCode = result;
+        }
+        return result;
+    }
+
+    @Override
+    @Safe
+    public String toString() {
+        return "SafeExternalLongExample{safeExternalLongValue: " + safeExternalLongValue
+                + ", optionalSafeExternalLong: " + optionalSafeExternalLong + '}';
+    }
+
+    public static SafeExternalLongExample of(@Safe long safeExternalLongValue, long optionalSafeExternalLong) {
+        return builder()
+                .safeExternalLongValue(safeExternalLongValue)
+                .optionalSafeExternalLong(Optional.of(optionalSafeExternalLong))
+                .build();
+    }
+
+    private static void validateFields(Optional<Long> optionalSafeExternalLong) {
+        List<String> missingFields = null;
+        missingFields = addFieldIfMissing(missingFields, optionalSafeExternalLong, "optionalSafeExternalLong");
+        if (missingFields != null) {
+            throw new SafeIllegalArgumentException(
+                    "Some required fields have not been set", SafeArg.of("missingFields", missingFields));
+        }
+    }
+
+    private static List<String> addFieldIfMissing(List<String> prev, Object fieldValue, String fieldName) {
+        List<String> missingFields = prev;
+        if (fieldValue == null) {
+            if (missingFields == null) {
+                missingFields = new ArrayList<>(1);
+            }
+            missingFields.add(fieldName);
+        }
+        return missingFields;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Generated("com.palantir.conjure.java.types.BeanBuilderGenerator")
+    public static final class Builder {
+        boolean _buildInvoked;
+
+        private long safeExternalLongValue;
+
+        private Optional<Long> optionalSafeExternalLong = Optional.empty();
+
+        private boolean _safeExternalLongValueInitialized = false;
+
+        private Builder() {}
+
+        public Builder from(SafeExternalLongExample other) {
+            checkNotBuilt();
+            safeExternalLongValue(other.getSafeExternalLongValue());
+            optionalSafeExternalLong(other.getOptionalSafeExternalLong());
+            return this;
+        }
+
+        @JsonSetter("safeExternalLongValue")
+        public Builder safeExternalLongValue(@Safe long safeExternalLongValue) {
+            checkNotBuilt();
+            this.safeExternalLongValue = safeExternalLongValue;
+            this._safeExternalLongValueInitialized = true;
+            return this;
+        }
+
+        @JsonSetter(value = "optionalSafeExternalLong", nulls = Nulls.SKIP)
+        public Builder optionalSafeExternalLong(@Nonnull Optional<? extends Long> optionalSafeExternalLong) {
```

Comment:
again, not sure if we want the parameter to have a safety annotation, or if we don't support wrapping external references in optional

Reply:
Ideally I think we'd end up with `Optional<@Safe ? extends Long>`

Reply:
Would we want `@Safe Optional<@Safe ? extends Long>` like [here](https://github.com/palantir/conjure-java/blob/develop/conjure-java-core/src/integrationInput/java/com/palantir/product/OptionalExample.java#L100)? Or would we omit the external annotation for external imports?

Reply:
I think the linked example should ideally only have a single `@Safe`:
```java
public Builder item(@Nonnull Optional<@Safe String> item) {
```

Reply:
For posterity:

Poet has a bug that will make handling containers of external imports hard (like optional<long>). We use [WildcardTypeName](https://github.com/palantir/conjure-java/blob/develop/conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderAuxiliarySettersUtils.java#L99-L129) to represent the wildcard type extending an external import. But when emitting, WildcardTypeName essentially [ignores its annotations](https://github.com/square/javapoet/blob/master/src/main/java/com/squareup/javapoet/WildcardTypeName.java#L64-L71), which I verified in this (extremely rough) [test](https://github.com/palantir/conjure-java/pull/1944/files#diff-cc2d5fa6d815eb2ce4ba4691ac1ceced1dd649598ba8400ab31cc3b1bced1b9bR202-R217). So in order to generate code that looks like "@Safe ? extends Long>" we would have to do something extremely hacky, like eschew javapoet and just stick in a string which doesn't feel......great.

So we decided to settle for the annotation on the wrapper itself, as in `@Safe Optional<...>`